File: examples/anim.py
#!/usr/bin/env python
# -*- no-plot -*-
"""
Compare line drawing performance of Matplotlib vs. GR

These are the results on an iMac 3,4 GHz Intel Core i7::

    fps (mpl):   29
    fps  (GR): 1002
      speedup:   34.6
"""

from pylab import plot, draw, pause
from numpy import arange, sin, pi
from time import time, sleep

x = arange(0, 2 * pi, 0.01)

tstart = time()
line, = plot(x, sin(x))
for i in arange(1, 200):
    line.set_ydata(sin(x + i / 10.0))
    draw()
    pause(0.0001)

fps_mpl = int(200 / (time() - tstart))
print('fps (mpl): %4d' % fps_mpl)

from gr.pygr import plot

tstart = time()
for i in arange(1, 200):
    plot(x, sin(x + i / 10.0))
    sleep(0.0001)

fps_gr = int(200 / (time() - tstart))
print('fps  (GR): %4d' % fps_gr)

print('  speedup: %6.1f' % (float(fps_gr) / fps_mpl))

File: examples/anim.jl
using PyPlot

x = [0:0.01:2*pi]

tic()
line, = plot(x, sin(x))
for i = 1:200
    line[:set_ydata](sin(x + i / 10.0))
    draw()
end

fps_gr = int(200 / toc())
println("fps  (PyPlot): ", fps_gr)

using jlgr

tic()
for i = 1:200
    plot(x, sin(x + i / 10.0))
end

fps_gr = int(200 / toc())
println("fps  (GR): ", fps_gr)

